src/oidcop/oidc/token.py

```diff
@@ -45,11 +45,16 @@ def process_request(self, req: Union[Message, dict], **kwargs):
         _session_info = _mngr.get_session_info_by_token(_access_code, grant=True)
         logger.debug(f"Session info: {_session_info}")
 
-        if _session_info["client_id"] != req["client_id"]:
-            logger.debug("{} owner of token".format(_session_info["client_id"]))
+        client_id = _session_info["client_id"]
```

Comment:
that's the same code of before, in oauth2/token.py ...


Mhh, can we have a decoupled function that returns a boolean (true it_can, false no_it_cannot ) for doing this check?

Reply:
There is a lot of duplicate code between these two files.

We probably need a small refactor.

Reply:
if you can do it in this PR it would be better, every left behind becomes then forgotten :)